### ISSUE TITLE
Very high processor and memory utilization after 4183

### ISSUE THREAD

**Issue by jsnyder886** [bitfocus/companion#2030]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I upgraded from 4180 to 4183 this morning on my windows 11 PC.  Have never been able to open companion immediately after upgrading.  Tried to roll back to 4180, 4177 and 4170 as these were the last three builds that I had that were working as expected.  Roll back has failed with the same results.  Companion will install but when run, processor is very high and memory climbs by the gb it seems.  Can not load web interface.

### Environment (please complete the following information)

```markdown
- OS: Windows 11
- Browser: Chrome 101.0.4951.67 (Official Build) (64-bit)
```

### Additional context

![Screenshot 2022-05-23 114932](https://user-images.githubusercontent.com/69740641/169858437-a9ea83eb-7d07-4fa5-8de5-8a6c3b8dd624.png)

**Comment by amcchord** [comment]
![Screen Shot 2022-05-23 at 4 05 22 PM](https://user-images.githubusercontent.com/9466882/169897488-90b9e999-c03c-4326-b162-e01ee945ecfb.png)
I too am seeing this issue.. but it looks more like a loop. It appears in long running sessions of companion.

**Comment by Julusian** [comment]
can someone provide me with a db file? if it is still happening after restarting.

**Comment by jsnyder886** [comment]
Just checking in, did you receive my db via email?

**Comment by jsnyder886** [comment]
So I've dug in a lot deeper into companion than I ever have before, and I was able to disable my propresenter module manually.  After doing so, companion starts up just fine and appears quite happy.  Except of course I cannot connect to propresenter.

I don't know how to troubleshoot beyond this point however.  Where do I go from here?

@amcchord I'm curious are you also using the propresenter module?

**Comment by Julusian** [comment]
@jsnyder886 I have thanks, but I havent had a chance to look into this yet

This is sounding a bit like #2022 and #1992, which all sounds similar but 'caused' by different modules.

Is there a companion.log file in the same place as the db? That might give some clues, or might not.

**Comment by Julusian** [comment]
Ive had this running for a couple of hours now, and there is no sign of cpu usage increasing, or socket leaking or anything like that.

It doesnt look like you have any triggers, so some kind of infinite loop starting is less likely but still possible.
This could be something like it only happens when pro-presenter is connected. Could you try having the propresenter module enabled, but with propresenter closed? If you then open it does it start having issues?
Perhaps if you leave that propresenter module disabled and configure a new one identically, does it still misbehave (ie, instance is there but no actions or feedbacks are using it)
Or after backup up your db, do a wipe of every page. perhaps one of the buttons showing the instance status is causing a loop?

Im feeling a bit stuck on how to debug this much further, except for adding a lot of temporary logging and hoping something gets logged way more than expected. But without knowing where to add logging, its likely to completely miss the issue..

**Comment by jsnyder886** [comment]
So late last night I discovered what appears to be an unhandled exception
in the ProPresenter module.  I had an announcement loop set within that
program that was empty and the companion module did not appear to being
okay with a presentation length of zero and my guess is there is something
to do with the division by zero. I was able to start and then stop that
module quickly and then dump a log file for grayshirtguy to see if there's
anything that he may be able to glean from it in hopes of preventing
something similar in the future. My guess is it's just a really weird
coincidence that I also happen to upgrade my companion installation at the
same time. I'm fine with you either moving this to the ProPresenter module
or closing it out.

Thank you for your willingness to help!
